xbmc/video/VideoDatabase.cpp

```diff
@@ -11701,17 +11746,25 @@ std::string CVideoDatabase::GetVideoItemTitle(VideoDbContentType itemType, int d
   }
 }
 
-void CVideoDatabase::InitializeVideoVersionTypeTable()
+void CVideoDatabase::InitializeVideoVersionTypeTable(int schemaVersion)
 {
   try
   {
     BeginTransaction();
 
     for (int id = VIDEO_VERSION_ID_BEGIN; id <= VIDEO_VERSION_ID_END; ++id)
     {
-      std::string type = g_localizeStrings.Get(id);
-      m_pDS->exec(PrepareSQL("INSERT INTO videoversiontype VALUES(%i, '%s', %i)", id, type.c_str(),
-                             VideoAssetTypeOwner::SYSTEM));
+      const std::string type{g_localizeStrings.Get(id)};
+      if (schemaVersion < 127)
+      {
+        m_pDS->exec(PrepareSQL("INSERT INTO videoversiontype VALUES(%i, '%s', %i)", id,
+                               type.c_str(), VideoAssetTypeOwner::SYSTEM));
```

Comment:
A better change would be to just replace the SQL with this:
`INSERT INTO videoversiontype (id, name, owner) VALUES (%i, '%s', %i)`
It's bad practice to not specify the column names. and causes issues when the table definition changes.

Reply:
Oh yes. I will change that.

Reply:
Hold on, you were right, a change is needed for the new install case, where no migration is going to take place.
So either what you did, or what I suggested + a DEFAULT value for the column.

Reply:
Sorry, I do not understand. Post a diff please.

Reply:
""INSERT INTO videoversiontype (id, name, owner, itemType) VALUES(NULL, '%s', %i, %i)" seems to work just fine. I actually copied that from existing code.

Reply:
Yes it works but that wasn't the issue.
For a brand new installation, the migration code assigning itemType = VideoAssetType::VERSION will not be executed so this function needs to provide the value during the INSERT. otherwise the field will be NULL.

Reply:
Not sure we have an issue here with ne install. I think I covered this case. Looking at CreateTables:

```c++
  m_pDS->exec("CREATE TABLE videoversiontype (id INTEGER PRIMARY KEY, name TEXT, owner INTEGER, "
              "itemType INTEGER)");
  InitializeVideoVersionTypeTable(GetSchemaVersion());
```

Important and new with this PR is passing the schema version number to `InitializeVideoVersionTypeTable`. Looking at changed implementation for `InitializeVideoVersionTypeTable`:

```c++
    for (int id = VIDEO_VERSION_ID_BEGIN; id <= VIDEO_VERSION_ID_END; ++id)
    {
      const std::string type{g_localizeStrings.Get(id)};
      if (schemaVersion < 127)
      {
        m_pDS->exec(
            PrepareSQL("INSERT INTO videoversiontype (id, name, owner) VALUES(%i, '%s', %i)", id,
                       type.c_str(), VideoAssetTypeOwner::SYSTEM));
      }
      else
      {
        m_pDS->exec(PrepareSQL(
            "INSERT INTO videoversiontype (id, name, owner, itemType) VALUES(%i, '%s', %i, %i)", id,
            type.c_str(), VideoAssetTypeOwner::SYSTEM, VideoAssetType::VERSION));
      }
    }
```

You see that itemType will be initialised properly, if the column is present (version > 127).

Reply:
yes that's fine.